renku/core/commands/cluster.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2017-2021 - Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Renku cluster command."""
+
+from pathlib import Path
+from subprocess import PIPE, STDOUT, run
+
+from jinja2 import Template
+
+from renku.core.incubation.command import Command
+from renku.core.management.config import CONFIG_LOCAL_PATH
+from renku.core.commands.docker import detect_registry_url
+
+CONFIG_SECTION = "cluster"
+IMAGE_SHA = 7
+
+
+def _prepare_cluster_config(client, sbatch_options, gitlab_token):
+    """Set up configuration for cluster execution"""
+    registry_url = detect_registry_url(client, auto_login=False)
+    # get the latest commit SHA at remote
+    renku_commit = client.repo.head.reference.tracking_branch().commit.hexsha
+    git_config = client.repo.config_reader()
+    cluster_config = {
+        # default
+        "sbatch_template": "sbatch_script_template.sh",
+        "sbatch_script": "sbatch_script.sh",
+        # user input
+        "sbatch_options": sbatch_options,
+        "gitlab_token": gitlab_token,
+        # read from repo
+        "git_username": git_config.get_value("user", "name", None),
+        "git_email": git_config.get_value("user", "email", None),
+        "git_remote": registry_url.href,
+        "sbatch_image": f"{registry_url.image}/{renku_commit[:IMAGE_SHA]}-batch",
```

Comment:
Is there a way that we can ensure that the image is built and exists on the remote? Currently this would depend on CI running, right?

Reply:
yes, it depends on CI running. we can call [GitLab container registry API](https://docs.gitlab.com/ee/api/container_registry.html) with the gitlab token to check if the image is there.